### ISSUE TITLE
CAT tool with macOS Sierra

### ISSUE THREAD

**Issue by robstoll87** [dokterdok/Continuity-Activation-Tool#312]
I installed the macOS Sierra beta and the CAT tool BETA for Sierra yet it doesn't seem to work. In "general" it had handoff checked.

Please include the following information.

- OS Version: 10.12 Beta (16A201w)
- Mac model: MacBook Pro (15-inch, Mid 2010)
- Dongle: IOGEAR Bluetooth 4.0 USB Micro Adapter (GBU521)

Any ideas on how I can get this to work? It worked for me in the past. 

Thanks,
Rob

**Comment by mwoolweaver** [comment]
Did you run it from the /Applications directory?

**Comment by robstoll87** [comment]
I dont believe so. Should I drag it to applications and try reinstalling it? Thanks for the quick response!

**Comment by robstoll87** [comment]
![screen shot 2016-06-27 at 11 48 30 pm](https://cloud.githubusercontent.com/assets/5217627/16403296/b11b31da-3cc1-11e6-9675-c6fd878497a0.png)

**Comment by mwoolweaver** [comment]
Yes

**Comment by robstoll87** [comment]
![screen shot 2016-06-28 at 12 15 38 am](https://cloud.githubusercontent.com/assets/5217627/16403684/c855cf28-3cc5-11e6-9064-674014488f8e.png)
Still doesn't work.

**Comment by mwoolweaver** [comment]
Do you have the options to enable Continuity/HandOff though in Preferences.app?

**Comment by robstoll87** [comment]
Yes

**Comment by mwoolweaver** [comment]
Ok so I had a weird issue were I didn't see any continuity features for at least 4 days (see issue #308) and then they just magically started working for no given reason...

**Comment by iPhonda** [comment]
Hello, I've a Mac mini (mid) 2011 and buy the Asus bt400, when installing everything goes fine then I restart the Mac but the HandOff does not appear on Preferences.app...
Whats wrong???

Ps : MacOs Sierra

**Comment by mactofou** [comment]
I installed on my macbook pro sierra 2009 with a patch and after I installed CAT! after the reboot no kernel panic but the mac still hang on the progress bar and restarts after 10 minutes. 😞

**Comment by hrubymar10** [comment]
Same for me…

MacBook Pro 13' Mid 2011 (MacBookPro7,1)
macOS Sierra 10.12 Beta (16A201w)
Asus BT-400 Dongle (worked in Yosemite, but in El Capitan not)

ma

**Comment by iPhonda** [comment]
On mine appear that option but on the lock screen he try's to unlock with the watch but after a 3 seconds ask the password again...

**Comment by robstoll87** [comment]
This is the report I get. Any idea why its still not working?
![screen shot 2016-07-17 at 8 52 05 pm](https://cloud.githubusercontent.com/assets/5217627/16904294/014af4c0-4c61-11e6-8071-8258caff6d17.png)

**Comment by maciboy** [comment]
Just installed the macOS Sierra GM on my MacBook Pro 15’’ 2010.
I’m using a Gmyle Dongle (https://www.amazon.de/gp/product/B007MKMJGO/ref=oh_aui_detailpage_o00_s00?ie=UTF8&psc=1)

CAT is working fine so fare, no panics, no unavailable Wifi or BT :)

Just one thing was a bit strange during installation: backups of kext files weren’t possible due to „read only filesystem“.
So no backups of IO kexts :(

**Comment by frizby66** [comment]
I Have had the same message ever since I installed Sierra and ran CAT.

**Comment by miguelmanalo** [comment]
@robstoll87 I did something and now Handoff is working!! 
https://imgur.com/a/CdzjU

Clipboard sharing, Safari/iMessage handoff!

What I did is:

Settings > iCloud > iCloud Drive > Toggle Off > Toggle On > Stuff works

NB: Before trying this I did do a Restore > Restore from Backup to 10.0.1. I don't know if this helped with anything. I did not touch anything on the macOS side.

- OS Versions: macOS Sierra 10.12 (16A320), iOS 10.0.1 (14A403)
- Mac model: MBA mid-2011, 11-inch, MacBookAir4,1
- iPhone model: iPhone 6S
- No dongle, no replaced card, using built-in WiFi, built-in Bluetooth

**Comment by mobile4** [comment]
Macbook 13' 2010 mid. Gmyle dongle. HDD has been replaced to SSD. Worked well with the latest Yosemite.
El Capitan - CAT installed without errors but kernel panic on reboot. Tried both version of CAT from Master and Beta branches.
Sierra - Some errors during installation of CAT. Something related to permissions and kext back up. Hangs on reboot.

1. Probably problem related to SSD? Everybody who has the same problem. Do you use SSD instead of HDD?
2. As I wrote above MBP hangs on reboot. Please make detailed instruction on how to replace wifi and bluetooth kext back to original. It is very boring to reinstall OS for the next experiment.

**Comment by fischersd** [comment]
What else is CAT doing for the ContinuitySupport patch?  I check my SystemParameters.plist (in frameworks) and there is an entry there for my board-ID (Mac Mini, late 2012) - Mac-031AEE4D24BFF0B1.
But when I run the CAT diags (after doing a force using the contitool.sh script), what you're checking still isn't patched:

--- Hardware/OS checks ---
Verifying Continuity status...          OK. OS X reports Continuity as active
Verifying Mac model reference...        OK. Known compatible Mac Model detected: Macmini6,1
Verifying Mac board-id...               OK. Long board id detected: Mac-031AEE4D24BFF0B1
Verifying OS X version...               Warning: This tool wasn't tested with OS X versions higher than 10.10. Detected OS version: 10.12
Verifying Wi-Fi hardware...             OK. A Broadcom AirPort card is active, and is using the Continuity compatible Brcm4360 kext
Verifying AWDL status...                OK. Wi-Fi hardware is compatible with AWDL, but no such interface is up. This tool can try to fix this.
Verifying Bluetooth hardware...         OK. The internal Bluetooth card is active
Verifying Bluetooth version...          OK. Bluetooth 4.0 detected
Verifying Bluetooth features...         OK. Bluetooth features are Continuity compliant
Verifying Bluetooth firmware...         OK. Bluetooth firmware version: v126 c9195
Verifying Login Item...                 OK. Login item for Auto Continuity Check is not set.

--- Modifications check ---
Verifying OS kext protection...         OK. Kext developer mode is already active
Verifying SIP...                        Ok. System Integrity Protection is already disabled
Verifying ContinuitySupport...          Print: Entry, ":Mac-031AEE4D24BFF0B1:ContinuitySupport", Does Not Exist
NOT OK. Unknown state. Your Mac might not be compatible.
Verifying kexts readability...          OK. Wi-Fi and Bluetooth kexts were found and could be read
Verifying Wi-Fi whitelist status...     OK. The whitelist is correctly patched with your board-id
Verifying BT4 dongles compatibility...  OK. Compatibility with BT4 USB dongles is not enabled, this tool can fix this if a dongle is plugged in
Verifying old Wi-Fi kext presence...    OK. Legacy Wi-Fi driver Brcm4331 was already removed
Verifying legacy Wi-Fi card patch...    OK. The patch is already done. Old Broadcom Wi-Fi cards may work.
--- Modifications check ---

(yes, it says the model is ok...did a little tinkering with the script - but not able to find what you're doing for this check).
